### PR TITLE
Add screen for updating the patient's prescription #6

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/889/add-screen-for-updating-the-patient-s-prescription

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoView.kt
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoView.kt:74]: The `currentModel` property on `MobiusDelegate` is deprecated since we added it only for the purposes of Mobius migration. Since we are done with that, I was thinking we can remove this property.

If we decide to keep it for cases like these, however, we might want to remove the deprecation notice.

File: app/src/main/res/layout/screen_teleconsult_prescription.xml
Comment (vinaysshenoy) [inline app/src/main/res/layout/screen_teleconsult_prescription.xml:54]: I'm curious, why do we need this additional container?

If we need the additional frame with a different colour background, maybe `FrameLayout` would be a better choice.

Comment (vinaysshenoy) [reply app/src/main/res/layout/screen_teleconsult_prescription.xml:54]: I'm also surprised that `ConstraintLayout` does not have a widget for this use case 😅

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]
Overall PR looks good, left a few comments.

File: app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoView.kt
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoView.kt:124]: Yes I will do this as part of `Refactoring SignatureActivity` [PR](https://github.com/simpledotorg/simple-android/pull/1744)

File: app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoEvent.kt
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoEvent.kt:23]: Yes, I was waiting for this PR to be merged in since I have added the analytics name in #1744

File: app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoView.kt
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoView.kt:144]: Ah true, will change that.

File: app/src/main/res/layout/view_teleconsult_doctor_info.xml
Comment (msasikanth) [reply app/src/main/res/layout/view_teleconsult_doctor_info.xml:57]: `addSignatureButton` is a outline button, this frame is supposed to emulate that. Even if we set the background manually it's not compulsory for both of these to have same background 🤔 .

File: app/src/main/res/layout/screen_teleconsult_prescription.xml
Comment (msasikanth) [reply app/src/main/res/layout/screen_teleconsult_prescription.xml:54]: Oh we can do that with `ConstraintLayout` it just requires usage of more widgets 😂 . So we can do something like setup barriers on top and bottom of the button and have a `View` with background color constrainted to those barriers.

File: app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoView.kt
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoView.kt:74]: Yep, I forgot to ask you about removing deprecation notice.

File: app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/TeleconsultRecord.kt
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/TeleconsultRecord.kt:49]: Waiting for the `TeleconsultRecord` sync resource to be added, currently we don't have `SyncStatus` in the object.

File: app/src/main/res/layout/view_teleconsult_doctor_info.xml
Comment (msasikanth) [reply app/src/main/res/layout/view_teleconsult_doctor_info.xml:57]: I am unsure about that, If we want to do that we can no longer use `MaterialButton`, because it doesn't draw background drawables. I can set a `Foreground` to emulate the stroke but it's only for API 23.

Also I am not sure how many times this get's changed TBH. Since these buttons are emulating the outline text fields used in the rest of the screen. Let's consider this later on?

WDYT?

Comment (vinaysshenoy) [reply app/src/main/res/layout/view_teleconsult_doctor_info.xml:57]: Makes sense.

File: app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/TeleconsultRecord.kt
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/TeleconsultRecord.kt:49]: Can we create a ticket then that lists all the different places that have to be updated to include the mark `syncStatus` as `PENDING` once the PR gets merged?

File: app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoView.kt
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/teleconsultlog/prescription/doctorinfo/TeleconsultDoctorInfoView.kt:74]: Let's remove it then.

File: app/src/main/res/layout/screen_teleconsult_prescription.xml
Comment (vinaysshenoy) [reply app/src/main/res/layout/screen_teleconsult_prescription.xml:54]: Right, should we use a `FrameLayout` then if all we want is just a background container?

File: app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/TeleconsultRecord.kt
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/TeleconsultRecord.kt:49]: Yep, will do that.

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/TeleconsultRecord.kt:49]: Tracking [here](https://app.clubhouse.io/simpledotorg/story/1278/add-syncstatus-for-teleconsultrecord)

**Review by vinaysshenoy — DISMISSED** [review]

**Review by vinaysshenoy — APPROVED** [review]

**Pull request merged**
